src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-server.cpp

```diff
@@ -40,11 +42,14 @@ using chip::DeviceLayer::GetDiagnosticDataProvider;
 
 namespace {
 
-class WiFiDiagosticsAttrAccess : public AttributeAccessInterface
+class WiFiDiagosticsGlobalInstance : public AttributeAccessInterface, public CommandHandlerInterface
```

Comment:
If we're refactoring this anyway, I think this should take the `EndpointId` and `DiagnosticDataProvider` as constructor parameters rather than adding more global singletons.

Reply:
This PR renames gAttrAccess to gWiFiDiagnosticsInstance without adding any new global singletons. We can merge this with the WiFiDiagnosticsServer singleton to reduce the total number of global instances. 

Rather than simply refactoring, this PR replaces the Ember-specific function with the existing generic AttributeAccessInterface and CommandHandlerInterface. 

Could you explain more how DiagnosticDataProvider could be used to implement the same functionality?.

Reply:
The code currently uses `DeviceLayer::GetDiagnosticDataProvider()` in the methods that access the DiagnosticDataProvider; it would be nicer to inject this as a constructor parameter instead as this would make the code unit testable and allow multiple independent cluster instances. Though in practice multiple Wi-Fi Diagnostic clusters probably don't make sense on any current device types.

You're right that it doesn't increase global singletons, so I guess any further refactoring would be just a "nice to have" at this point.

Reply:
Acked